### ISSUE TITLE
TF24 hydraulics: Leaf constructor does not return error when argument is mislabelled.

### ISSUE THREAD

**Issue by itowers1** [traitecoevo/plant#377]
A potential opportunity for error relating to Leaf constructor not causing an error when arguments are mislabelled. For example, in the simple water use model branch. The appropriate argument name for vcmax is `vcmax_25`. If we instead use `vcma`, the function surpisingly succeeds.

```
devtools::load_all()
sw <- FF16w_Strategy()
leaf <- Leaf(vcma= sw$vcmax, b = sw$b, c = sw$c, psi_crit = sw$psi_crit, epsilon_leaf = sw$epsilon_leaf, beta1 = sw$beta1, beta2 = sw$beta2, jmax_25 = sw$jmax_25, hk_s = sw$hk_s, a = sw$a, curv_fact_elec_trans = sw$curv_fact_elec_trans, curv_fact_colim = sw$curv_fact_colim)
leaf$set_physiology(PPFD = 1000, psi_soil = 0, leaf_specific_conductance_max = 4.8e-05, atm_vpd = 2, ca = 40, sapwood_volume_per_leaf_area = 1.8e-04, rho = 608, a_bio = 0.0245, leaf_temp = 25, atm_o2_kpa = 21, atm_kpa = 101.3)
#returns 96
leaf$vcmax_
```

Need to either enforce that constructor cannot have arguments which do not belong to it, or to inform user with a warning message.

**Comment by dfalster** [comment]
Hi @itowers1 

Can you confirm if this is still an issue in your latest implementation of TF24:leaf?

**Comment by itowers1** [comment]
Indeed, still an issue. Only passes without error if the arguments are mislabelled in the following manner "vcma" instead of "vcmax_25". Value assigned to "vcma" still correctly parameterises "vcmax_25" so I suppose its not really a big deal. Causes an error if the argument is labelled as "vcmax". `Error` in plant:::Leaf(vcmx = vcmax_25, jmax_25 = jmax_25, c = c, b = b,  : 
  unused argument (vcmx = vcmax_25)`

**Comment by dfalster** [comment]
Fixed in commit a8bd71f9 on branch `fix/issues-377-381`.

R's partial argument matching was silently resolving abbreviations like `vcma` to `vcmax_25`. The fix adds `R/zzz_leaf_wrapper.R`, which overrides the RcppR6-generated `Leaf()` constructor and inspects the raw call via `sys.call()` before partial matching applies — any argument name that is not an exact match now raises an error with a clear message listing the valid names.

A regression test was added to `tests/testthat/test-leaf.r`.
